components/Buttons/src/MDCFloatingButton.m

```diff
@@ -273,19 +341,168 @@ - (void)layoutSubviews {
   self.titleLabel.frame = UIEdgeInsetsInsetRect(self.titleLabel.frame, self.titleEdgeInsets);
 }
 
-#pragma mark - Subclassing
+#pragma mark - Property Setters/Getters
+
+- (void)setMode:(MDCFloatingButtonMode)mode {
+  BOOL needsShapeUpdate = _mode != mode;
+  _mode = mode;
+  if (needsShapeUpdate) {
+    [self updateShape:YES];
```

Comment:
This doesn't read very well, what is this boolean for? I'd change it to something like `updateShapeForcingResize`

Reply:
Done.